### PR TITLE
[NPU][A5] Supoort MXFP8-W8A8 for Ascend NPU

### PR DESCRIPTION
**Pull request by LinyuanLi0046**
## Motivation

Add Ascend NPU support for ModelSlim W8A8_MXFP8  path of sglang .

## Modifications

- Extend ModelSlimConfig to recognize W8A8_MXFP8 for both linear and MoE layers in the existing modelslim scheme dispatch path.
- Add new ModelSlim MXFP8 schemes:
-  MXFP8 linear layers
-  MXFP8 MOE layers

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** -- add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** -- `run-ci` is required first.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
     ModelSlimW4A8Int8MoE,
     ModelSlimW8A8Int8,
     ModelSlimW8A8Int8MoE,
+    ModelSlimW8A8MxFp8,
```
Comment (LinyuanLi0046) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:24]: The runtime capability requirement for this MXFP8 path is not clearly declared. Public Ascend MXFP8 documentation mostly describes the diffusion path and gives a minimum CANN requirement, while this PR adds an SRT/LLM ModelSlim path without version probing or actionable errors. Please document the minimum CANN/torch_npu versions and add a backend initialization check for the required dtype and ops

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py [hunk 2]
```diff
@@ -18,6 +23,53 @@ def __init__(
...
+        layer.weight.data = layer.weight.data.transpose(-1, -2).contiguous()
+        layer.weight.data = npu_format_cast(layer.weight.data)
+        weight_scale = layer.weight_scale.data.transpose(-1, -2).contiguous()
+        k32, n = weight_scale.shape
+        weight_scale = (
+            weight_scale.view(k32 // 2, 2, n).permute(0, 2, 1).contiguous()
+        )
+        layer.weight_scale.data = weight_scale
+
+    def apply(
```
Comment (LinyuanLi0046) [inline python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:84]: NPUW8A8MxFp8LinearMethod.apply() accepts a bias argument but does not pass it to npu_quant_matmul

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py [hunk 3]
```diff
@@ -18,6 +23,53 @@ def __init__(
...
 
+class NPUW8A8MxFp8LinearMethod(_NPULinearMethodBase):
+
+    def process_weights_after_loading(self, layer: torch.nn.Module):
+        layer.weight.data = layer.weight.data.transpose(-1, -2).contiguous()
+        layer.weight.data = npu_format_cast(layer.weight.data)
+        weight_scale = layer.weight_scale.data.transpose(-1, -2).contiguous()
+        k32, n = weight_scale.shape
+        weight_scale = (
+            weight_scale.view(k32 // 2, 2, n).permute(0, 2, 1).contiguous()
```
Comment (LinyuanLi0046) [inline python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:80]: The scale reordering logic depends on k32 // 2 and view(..., 2, n) without explicit assertions. If the exported checkpoint block-scale layout differs from this assumption, the code may fail in view() with an unhelpful error. Please assert k32 % 2 == 0, validate the final shape against the expected quant_model_description.json schema, and include both original and expected shapes in the error message.

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py (hunk 1 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:24]: doc will be updated later

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py (hunk 3 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:80]: k32 % 2 == 0 is alwyas true for mxfp8 weight

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py (hunk 2 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:84]: now only support no bias
